src/hevm/src/EVM.hs

```diff
@@ -2012,6 +2011,79 @@ cheatActions =
                 assign (state . memory . word256At outOffset) res
           _ -> vmError (BadCheatCode sig),
 
+      action "file(bytes32,address,uint256)" $
+        \sig _ _ input -> case decodeStaticArgs input of
+          [what', addr, val] -> forceConcrete what' $ \(C _ (utf8Word -> what)) ->
+            makeUnique addr $ \(C _ (num -> usr)) -> fetchAccount usr $ \_ ->
+              case what of
+                "nonce" -> forceConcrete val $ \v -> assign (env . contracts . ix usr . nonce) v
+                "balance" -> forceConcrete val $ \v -> assign (env . contracts . ix usr . balance) v
+                _ -> vmError (BadCheatCode sig)
+          _ -> vmError (BadCheatCode sig),
+
+      action "file(bytes32,address)" $
+        \sig _ _ input -> case decodeStaticArgs input of
+          [what', addr] -> forceConcrete what' $ \(C _ (utf8Word -> what)) ->
+            case what of
+              "caller" -> assign (state . caller) (SAddr . sFromIntegral . rawVal $ addr)
```

Comment:
@MrChico I'm a little unclear on the details of the `frames` stack? do we need to change this anywhere else? I'm worried that this is going to mess up returning from the current frame?

Reply:
You should see how this is handled in `finishFrame`. When returning from a call, we simply install the last callframe https://github.com/dapphub/dapptools/blob/master/src/hevm/src/EVM.hs#L2293

Reply:
ok, so afaict this seems to be ok as is?

Reply:
yeah, it seems to be the expected behaviour to not modify the outer frame